### DISCUSSION TITLE
Request for Customizing `ondataavailable` in RecordPlugin

### DISCUSSION THREAD

**Discussion by gpgun0** [katspaugh/wavesurfer.js#3824] [Q&A] [answered]
First of all, thank you for providing such an excellent library. I have been using it extensively and it has been a great help in my projects.

I am currently facing the following issue:

I am using the wavesurfer library along with the `RecordPlugin`. I connect the `stream` returned by `startMic` to an audio context for processing, and then send it to the backend via a socket.

However, I encountered a problem specifically with iOS Chrome browsers. When the screen is locked and the browser enters background mode, the socket connection remains active, but the aforementioned process stops functioning. I understand that this is due to iOS's policy, which we cannot bypass.

Fortunately, I discovered that the `MediaRecorder` still retains resources in background mode, allowing us to use `ondataavailable` to send blobs through the socket.

With this in mind, it would be incredibly helpful if there were a feature that allows us to customize the `ondataavailable` event.

I wanted to ask if there is a particular reason why customizing `ondataavailable` is not currently supported. If not, would it be possible for me to submit a PR to add this feature?

Thank you for your time, and I look forward to your response. Have a great day!

Best regards.

**Comment by katspaugh** [comment, marked as answer]
I'm not sure I understand what you mean by customizing `ondataavailable` but happy to review a PR.
